packages/react/src/components/Table/tableReducer.js

```diff
@@ -212,17 +222,46 @@ export const filterSearchAndSort = (
   columns,
   advancedFilters = []
 ) => {
-  const { columnId, direction } = sort;
-
   const { value: searchValue } = search;
   const filteredData = filterData(data, filters, columns, advancedFilters);
   const searchedData =
     searchValue && searchValue !== '' ? searchData(filteredData, searchValue) : filteredData;
-  return !isEmpty(sort)
-    ? getCustomColumnSort(columns, columnId)
-      ? getCustomColumnSort(columns, columnId)({ data: searchedData, columnId, direction })
-      : getSortedData(searchedData, columnId, direction)
-    : searchedData;
+
+  if (isEmpty(sort)) {
+    return searchedData;
+  }
+
+  if (Array.isArray(sort)) {
+    let sortStack = firstBy(() => 0);
```

Comment:
`firstBy(() => 0);` seems a bit odd, is so that you can directly use `thenBy` in the loop below?

Perhaps this multisort can be broken out into its own function in order to improve readability for the 3 return statements of this larger filterSearchAndSort function.

Reply:
added a comment to explain the firstBy and moved multisort to its own function.